### PR TITLE
Support different working directories for data

### PR DESCRIPTION
**Pull request by guilherme-fh**
# PR Details

* Adds `chooseFolder` method to `Utils` to give support to directory selection
* Defines `workingDirectory` that is defined according to what's set on the `dataFolder.json` (or else defaults to the root directory). Upon setting a new `workingDirectory` the `dataFolder.json` is updated accordingly
* Adds a section to the Options screen to display the working directory and a button to change said directory which updates `workingDirectory`

## Description

This PR intends to give support to different working directories to access the data and test files.
Its purpose is to allow git synchronisation of these files, a feature that useful when working with teams.

## Motivation and Context

This PR intends to ease synchronisation of data and test files among large (or even small) teams

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by rodrigowoulddo — APPROVED** [review]

File: src/Mockingbird/Utils/Default.swift [hunk 1]
```diff
@@ -24,10 +25,51 @@ final class Default {
...
+        static var data: String { workingDirectory + "/data" }
+        static var test: String { workingDirectory + "/test" }
+
+        static var workingDirectory = ""
+
+        static var savedWorkingDirectory: String {
+
+            get {
+
+                guard let file = try? File(path: "\(Default.Folder.main)/configurations.json"),
```
Comment (erickjung) [inline src/Mockingbird/Utils/Default.swift:40]: should we use codable here?

File: src/Mockingbird/Utils/Default.swift [hunk 2]
```diff
@@ -24,10 +25,51 @@ final class Default {
...
+                }
+
+                return String(fileData.suffix(from: startRange))
+            }
+
+            set {
+
+                if let configurationsFile = try? File(path: "\(Default.Folder.main)/configurations.json") {
+
+                    let configurations = Configuration(dataFolder: newValue)
```
Comment (erickjung) [inline src/Mockingbird/Utils/Default.swift:54]: can we move those common codes inside if/else to a common function?

File: src/Mockingbird/UI/View/OptionsView.swift [hunk 3]
```diff
@@ -15,13 +15,28 @@ public class OptionsView {
...
                 }
                 NewLine()
+
+                Text("WORKING DIRECTORY: \(Default.Folder.workingDirectory)")
+                LargeButton("CHANGE DIRECTORY") {
+
+                    if let newDirectory = Util.chooseFolder() {
+
+                        Default.Folder.savedWorkingDirectory = newDirectory
+                        Default.Folder.workingDirectory = newDirectory
```
Comment (erickjung) [inline src/Mockingbird/UI/View/OptionsView.swift:33]: can we move this set to be inside the savedWorkingDirectory?

File: src/Mockingbird/Utils/Default.swift (hunk 1 above)
Comment (guilherme-fh) [reply src/Mockingbird/Utils/Default.swift:40]: Used codable after retrieving the data from the file. Take a new look!

File: src/Mockingbird/UI/View/OptionsView.swift (hunk 3 above)
Comment (guilherme-fh) [reply src/Mockingbird/UI/View/OptionsView.swift:33]: Moved!

File: src/Mockingbird/AppDelegate.swift [hunk 4]
```diff
@@ -20,6 +20,8 @@ class AppDelegate: NSObject, NSApplicationDelegate {
 
         Tracking.initialize()
 
+        Default.Folder.workingDirectory = Default.Folder.savedWorkingDirectory
```
[the remark this answers is not in the document]
Comment (guilherme-fh) [reply src/Mockingbird/AppDelegate.swift]: Moved!

File: src/Mockingbird/Utils/Default.swift (hunk 2 above)
Comment (guilherme-fh) [reply src/Mockingbird/Utils/Default.swift:56]: Moved to a new static function!

**Review by erickjung — APPROVED** [review]

**Pull request merged**